.github/workflows/sast.yml

```
@@ -0,0 +1,51 @@
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+
+    - name: Setup JDK 17
+      uses: actions/setup-java@v2
```

actionlint

the runner of "actions/setup-java@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)